pkg/build/builder/daemonless.go

```diff
@@ -313,10 +301,77 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 		DropCapabilities:        dropCapabilities(),
 	}
 
-	_, _, err := imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
+	_, _, err = imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
 	return err
 }
 
+func generateTransientMounts() ([]string, error) {
+	mounts := []string{}
+	mounts, err := appendRHSMMount(mounts)
+	if err != nil {
+		return mounts, err
+	}
+	mounts, err = appendCATrustMount(mounts)
+	if err != nil {
+		return mounts, err
+	}
+	return mounts, nil
+}
+
+func appendRHSMMount(mounts []string) ([]string, error) {
+	st, err := os.Stat("/run/secrets/rhsm")
+	if err != nil {
+		return mounts, errors.Wrap(err, "failed to stat /run/secrets/rhsm")
+	}
```

Comment:
@gabemontero this is a refactor of your original code. However, I think we should not return an error.

From our experience, consuming entitlements in builds is more of a "secondary" use case. The worst case is if a build does a `yum install` and we don't provide the entitlement keys, the build fails at yum install time. I don't think we should fail all builds because we can't set up the rhsm data.

Reply:
sure I'm good with that @adambkaplan